Project.Json.Test/JsonTest.cs

```diff
@@ -0,0 +1,200 @@
+            }
+            finally
+            {
+                File.Delete(jsonFile);
+            }
+        }
+
+        [TestMethod]
+        public void Arguments_WithInvalidJSONArgument()
+        {
+            //arrange
+            string jsonFile = CreateTempFile(new string[] { "[{'Name':'Red Apple'},{ 'Name':'Green Apple'}]" });
+
+            //assert
+            try
+            {
+                Assert.ThrowsException<ArgumentException>(() => Program.Main(new string[] { "json", jsonFile }));
+            }
+            finally
+            {
+                File.Delete(jsonFile);
+            }
+        }
+
+        [TestMethod]
+        public void Arguments_WithMissingJSONArgument() => Assert.ThrowsException<ArgumentException>(() => Program.Main(Array.Empty<string>()));
+
+        [TestMethod]
+        public void Arguments_WithTooManyArguments()
+        {
+            //arrange
+            string jsonFile = CreateTempFile(new string[] { "[{'Name':'Red Apple'},{ 'Name':'Green Apple'}]" });
+
+            try
+            {
+                //act
+                Program.Main(new string[] { "-json", jsonFile, "-test1", jsonFile, "-test2", jsonFile });
+            }
+            catch (ArgumentException aex)
+            {
+                //assert
+                Assert.Fail(aex.Message);
+            }
+            finally
+            {
+                File.Delete(jsonFile);
+            }
+        }
+
+        [TestMethod]
+        public void Arguments_WithInvalidJSONFileName() => Assert.ThrowsException<ArgumentException>(() => Program.Main(new string[] { "-json", "BadFile.txt" }));
+
+        [TestMethod]
+        public void File_WithValidJSONFormat()
+        {
+            //arrange
+            IFileHandling ifhJson = new JsonParser();
+
+            //assert
+            Assert.IsTrue(ifhJson.GetParsedData("[{'Name':'Red Apple'},{ 'Name':'Green Apple'}]").Length > 0);
+        }
+
+        [TestMethod]
+        public void File_WithInvalidJSONFormat()
+        {
+            //arrange
+            IFileHandling ifhJson = new JsonParser();
+
+            //assert
+            Assert.ThrowsException<JsonSerializationException>(() => ifhJson.GetParsedData("[{'Name':Red Apple},{'Name':Green Apple}]"));
+        }
+
+        [TestMethod]
+        public void File_WithValidJSONNonNumericDataSorted()
+        {
+            //arrange
+            IFileHandling ifhJson = new JsonParser();
+
+            //create json string
+            string json = "[{'Name':'Red Apple'},{'Name':'Green Apple'},{'Name':'Granny Smith Apple'},{'Name':null}]";
+
+            //create the expected results in the expected sort order
+            string[] expectedResults = new string[] { "Granny Smith Apple","Green Apple", "Red Apple",null };
+
+            //act
+            //create the testresults sorted with nulls at the end
+            string[] testResults = Program.GetSortedData(ifhJson.GetParsedData(json)).ToArray();
```

Comment:
`Scope`: `GetSortedData` functionality is part of the console (not the parser) and should be tested in `Project.ConsoleApp.Test` via `Program.Main()`.